### PR TITLE
Wire clusterclient.Client to AWSClusterConfig resource

### PR DESCRIPTION
**Pull request by tuommaki**
In order to be able to fetch version bundles for given release, configure &
wire up clusterclient.Client to AWSClusterConfig resource.

### CODE REVIEW

File: helm/cluster-operator-chart/templates/configmap.yaml [hunk 1]
```diff
@@ -23,6 +23,8 @@ data:
           certificate:
             ttl: '{{ .Values.Installation.V1.Auth.Vault.Certificate.TTL }}'
     service:
+      cluster:
+        address: 'http://cluster-service:8000'
```
Comment (tuommaki) [inline helm/cluster-operator-chart/templates/configmap.yaml]: Somehow my flag lookup returns empty string for this. Debugging...

Comment (xh3b4sd) [reply helm/cluster-operator-chart/templates/configmap.yaml]: Why do we even need that? Whatever is reconciled has the version bundle version we are interested in already. What do we need to ask `cluster-service` for?

Comment (tuommaki) [reply helm/cluster-operator-chart/templates/configmap.yaml]: We have release version, but we don't have version bundles and we need to fetch them.

Comment (tuommaki) [reply helm/cluster-operator-chart/templates/configmap.yaml]: Hmm... Now that I was actually taking a look, we need to change our provider extension. It does indeed have version bundle there, but that doesn't work for `cluster-operator` in any way. We should probably change it to be release version + version bundles or at least release version.

https://github.com/giantswarm/apiextensions/blob/master/pkg/apis/cluster/v1alpha1/aws_cluster_spec_types.go#L67

vs.

https://github.com/giantswarm/apiextensions/blob/master/pkg/apis/core/v1alpha1/cluster_guest_types.go#L14

File: main.go [hunk 2]
```diff
@@ -118,6 +118,7 @@ func mainWithError() error {
 	daemonCommand.PersistentFlags().String(f.Guest.Cluster.Calico.Subnet, "", "Network address for the CIDR block used by Calico.")
 	daemonCommand.PersistentFlags().String(f.Guest.Cluster.Kubernetes.API.ClusterIPRange, "", "CIDR Range for Pods in cluster.")
 	daemonCommand.PersistentFlags().String(f.Guest.Cluster.Vault.Certificate.TTL, "", "Vault certificate TTL.")
+	daemonCommand.PersistentFlags().String(f.Service.Cluster.Address, "http://localhost:8089", "http://<host>:<port> of the cluster service.")
```
Comment (kopiczko) [inline main.go]: IMO this should be `f.Service.ClusterService.Address`. My understanding is `.Service` key is this "service" configuration. Just to make sure it doesn't collide with things like `.Server`. `Cluster.Address` makes me think it's kubernetes address.

Comment (tuommaki) [reply main.go]: Ack. I'll change this. I copied this directly from `api`, but I guess that's not exactly up-to-date there :slightly_smiling_face:

**Comment by tuommaki** [comment]
> If this is a temp hack we probably don't need a flag at all. Const should do, but I don't mind.

Yeah, I thought about that, but then again, if there's any chance that we want to test this with something unexceptional (like locally or whatever), it's nice to have. It's also easy to tear down so I'm going with this :slightly_smiling_face:

**Pull request merged**